helper_funcs.inc.php

```diff
@@ -51,70 +53,141 @@ function pwg_privacy_reject_access ($msg) {
 	exit();
 }
 
-function pwg_privacy_serve_file ($path) {
-	global $conf;
+function pwg_privacy_get_mime_type($file_path) {
+    // retrieve the mime type for the file. This is based on code found on php.net
+	$arrayZips = array("application/zip", "application/x-zip", "application/x-zip-compressed");
 
-	$ext = get_extension($path);
-	$ext = strtolower($ext);
+	$arrayExtensions = array(".pptx", ".docx", ".dotx", ".xlsx");
 
-	if (!in_array($ext, $conf['file_ext'])) {
-		pwg_privacy_reject_access('File extension is not allowed');
+	$original_extension = (false === $pos = strrpos($file_path, '.')) ? '' : substr($file_path, $pos);
+
+	$finfo = new finfo(FILEINFO_MIME);
+
+	$type = $finfo->file($file_path);
+
+	if (in_array($type, $arrayZips) && in_array($original_extension, $arrayExtensions))
+	{
+		return $original_extension;
```

Comment:
Why do you return the extension here and not the mime type ?

Reply:
This code is heavily based on code I found on the internet. Using the mime type of the file is inappropriate since it is a zip file. So, you want to return a mime type that identifies it as an ms word document or excel document etc. I'm not sure if anyone actually serves files of this type, but it didn't seem to harm by leaving it in.

Reply:
From what I recall, the place I got this from said there is no mime type specific to e.g. an ms excel or ms word file, hence just use the file extension as that. Whether that is true or not I did not check.

Reply:
I don't think it makes sense for a method that suppose to return the mime type to return the extension instead.
I'm not sure what the issue with zips is, but I doubt if it's relevant to what this plugin does.
If you can find the source of this code and the reason for it we can reconsider.